src/science/misc/cable_climate.F90

```diff
@@ -15,271 +15,475 @@
+    climate%cs_shade(:,nsd)     = real(canopy%cs_sh) ! ppm
+
+    climate%scalex_sun(:,1:nsd-1) = climate%scalex_sun(:,2:nsd)
+    climate%scalex_sun(:,nsd)     = rad%scalex(:,1)
+
+    climate%scalex_shade(:,1:nsd-1) = climate%scalex_shade(:,2:nsd)
+    climate%scalex_shade(:,nsd)     = rad%scalex(:,2)
+
+    if (mod((ktau-kstart+1), ktauday) == 0) then  ! end of day
        ! get month and check if end of month
-       IsLastDay = .FALSE.
-       MonthDays = (/31,28,31,30,31,30,31,31,30,31,30,31/)
-       IF (LOY==366) MonthDays(2) = MonthDays(2) + 1
+       IsLastDay = .false.
+       MonthDays = (/31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31/)
+       if(LOY==366) MonthDays(2) = MonthDays(2) + 1
        nmonth = 1
        tmp = MonthDays(1)
-       DO WHILE(tmp.LT.idoy)
+       do while(tmp.lt.idoy)
           tmp = tmp +  MonthDays(nmonth+1)
           nmonth = nmonth + 1
-       ENDDO
+       enddo
 
-       IF (idoy == SUM(MonthDays(1:nmonth))) IsLastDay = .TRUE.
+       if (idoy == sum(MonthDays(1:nmonth))) IsLastDay = .true.
 
        ! On first day of year ...
-       IF (idoy==1) THEN
-
+       if (idoy==1) then
           ! ... reset annual GDD5 counter
-          climate%agdd5=0.0
-          climate%agdd0=0.0
-          climate%evap_PT = 0     ! annual PT evap [mm]
-          climate%aevap  = 0      ! annual evap [mm]
+          climate%agdd5   = 0.0
+          climate%agdd0   = 0.0
+          climate%evap_PT = 0.0 ! annual PT evap [mm]
+          climate%aevap   = 0.0 ! annual evap [mm]
+          ! ... reset annual min and max soil moisture
+          climate%dmoist_min = climate%dmoist
+          climate%dmoist_max = climate%dmoist
+          ! ... reset annual precip
+          climate%aprecip = 0.0
+          ! ...reset annual max nesterov index
+          climate%Nesterov_ann_max = 0.0
+          climate%fapar_ann_max = 0.0
+       endif
+
+       where ( ((patch%latitude>=0.0) .and. (idoy==COLDEST_DAY_NHEMISPHERE)) .or. &
+            ((patch%latitude<0.0) .and. (idoy==COLDEST_DAY_SHEMISPHERE)) )
+          ! In midwinter, reset GDD counter for summergreen phenology
+          ! climate%gdd5=0.0
+          climate%gdd0 = 0.0
+          ! reset day degree sum related to spring photosynthetic recovery
+          climate%gdd0_rec = 0.0
+       end where
+
+       where ( ((patch%latitude<=0.0) .and. (idoy==COLDEST_DAY_NHEMISPHERE)) .or. &
+            ((patch%latitude>0.0) .and. (idoy==COLDEST_DAY_SHEMISPHERE)) )
+          ! In mid-summer, reset dormancy fraction
+          climate%fdorm = 1.0
+       ENDWHERE
 
+       ! Update GDD counters and chill day count
+       climate%gdd0  = climate%gdd0  + max(0.0, climate%dtemp-0.0)
+       climate%agdd0 = climate%agdd0 + max(0.0, climate%dtemp-0.0)
 
-       ENDIF
+       climate%gdd5  = climate%gdd5  + max(0.0, climate%dtemp-5.0)
+       climate%agdd5 = climate%agdd5 + max(0.0, climate%dtemp-5.0)
 
-       !jhan: see CABLE Ticket#149 and above CABLE_LSM: comment
```

Comment:
May need to retain this `#ifndef` condition - I don't know why it was necessary and likely superceded by the  `#ifdef UM_CBL`